yoyodyne/train.py

```diff
@@ -165,6 +166,20 @@ def get_model_from_argparse_args(
         models.BaseEncoderDecoder.
     """
     model_cls = models.get_model_cls(args.arch)
+    if (
+        args.arch
+        in [
+            "pointer_generator_lstm",
+            "pointer_generator_transformer",
+            "transducer",
+        ]
+        and not args.tie_embeddings
+    ):
+        msg = f"--tie_embeddings set to {args.tie_embeddings}, but "
```

Comment:
if you're doing this though, better to make it a multiple lines of strings with implicit concatenation. String `+=` is really slow and happens at "runtime", but doing it like like:

```python
raise Error(
   "message part 1"
   "message part 2"
)
```

is handled once at "compile" (interpretation) time, not repeatedly at runtime. (And even interpretation can be cached, that's what `.pyc` and `.pyo` objects are.)

Reply:
Ah, good to know!!!

Reply:
Also, I had originally tried to handle this in the classes but had some issues. I think when we figure out a PR for class callbacks as you suggested in #111 , this can go there.